internal/controller/controllers/clusterdeployments_controller_test.go

```diff
@@ -1224,6 +1225,31 @@ var _ = Describe("cluster reconcile", func() {
 			Expect(result).Should(Equal(ctrl.Result{}))
 		})
 
+		DescribeTable("deleting AgentClusterInstall resource cancels the installation",
+			func(clusterStatus string) {
+				backEndCluster := &common.Cluster{
+					Cluster: models.Cluster{
+						ID:     &sId,
+						Status: swag.String(models.ClusterStatusInstalling),
+					},
+				}
+				mockInstallerInternal.EXPECT().GetClusterByKubeKey(gomock.Any()).Return(backEndCluster, nil).Times(2)
+				mockInstallerInternal.EXPECT().DeregisterClusterInternal(gomock.Any(), gomock.Any()).Return(nil)
+				mockInstallerInternal.EXPECT().CancelInstallationInternal(gomock.Any(), gomock.Any()).Return(backEndCluster, nil).Times(1)
+
+				simulateACIDeletionWithFinalizer(ctx, c, aci)
+				request := newClusterDeploymentRequest(cd)
+				result, err := cr.Reconcile(ctx, request)
+				Expect(err).ShouldNot(HaveOccurred())
+				Expect(result).Should(Equal(ctrl.Result{}))
+			},
```

Comment:
Would it be possible to check the cluster's status here to verify it's canceled?

Reply:
We're mocking the call to `CancelInstallationInternal` so there's not even a real DB backing this test so I don't think so.